tests/unit/hooks/test_zabbix.py

```diff
@@ -0,0 +1,194 @@
+from enum import Enum
+
+from flexmock import flexmock
+
+import borgmatic.hooks.monitor
+from borgmatic.hooks import zabbix as module
+
+server = 'https://zabbix.com/zabbix/api_jsonrpc.php'
+itemid = 55105
+username = 'testuser'
+password = 'fakepassword'
+api_key = 'fakekey'
+host = 'borg-server'
+key = 'borg.status'
+value = 'fail'
+
+data_host_key = {
+    "jsonrpc": "2.0",
+    "method": "history.push",
+    "params": {"host": host, "key": key, "value": value},
+    "id": 1,
+}
+
+data_itemid = {
+    "jsonrpc": "2.0",
+    "method": "history.push",
+    "params": {"itemid": itemid, "value": value},
+    "id": 1,
+}
+
+data_user_login = {
+    "jsonrpc": "2.0",
+    "method": "user.login",
+    "params": {"username": username, "password": password},
+    "id": 1,
+}
+
+auth_headers_api_key = {
+    'Content-Type': 'application/json-rpc',
+    'Authorization': f'Bearer {api_key}'
+}
+
+auth_headers_username_password = {
+    'Content-Type': 'application/json-rpc'
+}
+
+def test_ping_monitor_config_with_api_key_only():
+    hook_config = {
+        'api_key': api_key
+    }
+    flexmock(module.logger).should_receive('warning').once()
+
+    module.ping_monitor(
+        hook_config,
+        {},
+        'config.yaml',
+        borgmatic.hooks.monitor.State.FAIL,
+        monitoring_log_level=1,
+        dry_run=False,
+    )
+
+def test_ping_monitor_config_with_host_only():
+    hook_config = {
+        'host': host
+    }
+    flexmock(module.logger).should_receive('warning').once()
+
+    module.ping_monitor(
+        hook_config,
+        {},
+        'config.yaml',
+        borgmatic.hooks.monitor.State.FAIL,
+        monitoring_log_level=1,
+        dry_run=False,
+    )
+
+def test_ping_monitor_config_with_key_only():
+    hook_config = {
+        'key': key
+    }
+    flexmock(module.logger).should_receive('warning').once()
+
+    module.ping_monitor(
+        hook_config,
+        {},
+        'config.yaml',
+        borgmatic.hooks.monitor.State.FAIL,
+        monitoring_log_level=1,
+        dry_run=False,
+    )
+
+def test_ping_monitor_config_with_server_only():
+    hook_config = {
+        'server': server
+    }
+    flexmock(module.logger).should_receive('warning').once()
+
+    module.ping_monitor(
+        hook_config,
+        {},
+        'config.yaml',
+        borgmatic.hooks.monitor.State.FAIL,
+        monitoring_log_level=1,
+        dry_run=False,
+    )
+
+def test_ping_monitor_config_user_password_no_zabbix_data():
+    hook_config = {
+        'server': server,
+        'username': username,
+        'password': password
+    }
+    flexmock(module.logger).should_receive('warning').once()
+
+    module.ping_monitor(
+        hook_config,
+        {},
+        'config.yaml',
+        borgmatic.hooks.monitor.State.FAIL,
+        monitoring_log_level=1,
+        dry_run=False,
+    )
+
+def test_ping_monitor_config_api_key_no_zabbix_data():
```

Comment:
IMO you don't need to test *every* possible permutation of inputs. It's probably sufficient if you just had:

1. One test for each of the auth permutations.
2. One test for each of the Zabbix data / itemid permutations.

But I don't think you need a test for every cross product of those two areas. My rationale is that it's unlikely for there to be a bug involving both the auth _and_ the Zabbix data. Of course, if that turns out to not be the case down the road, we can always add a test then.

I don't feel super strongly.. Just trying to cut down on your test load!

Reply:
Hi, here is my 2 latest commits which should have all the tests required (hopefully): 

- https://github.com/borgmatic-collective/borgmatic/commit/e52e29444f04e4708692e2da4a5687069562d508
- https://github.com/borgmatic-collective/borgmatic/commit/fa6a4734d4375c01cb0a61315db7874145afd993

Here are the test scenarios I've included overall:

- missing data in the hook_config should produce a warning and return
- auth via `USERNAME`/`PASSWORD` while publishing data via `ITEMID`
- auth via `USERNAME`/`PASSWORD` while publishing data via `HOST`/`KEY`
- auth via `API_KEY` while publishing data via `ITEMID`
- auth via `API_KEY` while publishing data via `HOST`/`KEY`